concrete/src/Page/Sitemap/SitemapWriter.php

```diff
@@ -292,15 +292,20 @@ public function generate(callable $pulse = null)
                     $pulse($element);
                 }
                 if ($dispatchElementReady) {
-                    $this->director->dispatch(static::EVENTNAME_ELEMENTREADY, new GenericEvent(['sitemapPage' => $element]));
+                    $event = new GenericEvent(['sitemapPage' => $element], ['sitemapPage' => $element]);
+                    $this->director->dispatch(static::EVENTNAME_ELEMENTREADY, new GenericEvent(['sitemapPage' => $element], ['sitemapPage' => $element]));
+                    $element = $event->getArgument('sitemapPage');
                 }
                 if ($mode === static::MODE_HIGHMEMORY) {
                     if ($element instanceof SitemapHeader) {
                         $xmlDocument = $element->toXmlElement();
                     } else {
                         $xmlNode = $element->toXmlElement($xmlDocument);
                         if ($dispatchPageReadyDeprecated && $element instanceof SitemapPage) {
-                            $this->director->dispatch(static::EVENTNAME_PAGEREADY_DEPRECATED, new GenericEvent(['page' => $element->getPage(), 'xmlNode' => $xmlNode]));
+                            $eventArgs = ['page' => $element->getPage(), 'xmlNode' => $xmlNode];
```

Comment:
It's probably worth not keeping a copy of this in memory, so either unsetting it or passing the array in as a parameter would be good